sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproxCountDistinctForIntervals.scala

```diff
@@ -153,13 +129,14 @@ case class ApproxCountDistinctForIntervals(
       // endpoints are sorted into ascending order already
       if (endpoints.head > doubleValue || endpoints.last < doubleValue) {
         // ignore if the value is out of the whole range
-        return
+        return buffer
       }
 
       val hllppIndex = findHllppIndex(doubleValue)
-      val offset = mutableAggBufferOffset + hllppIndex * numWordsPerHllpp
-      hllppArray(hllppIndex).update(buffer, offset, value, child.dataType)
+      val offset = hllppIndex * numWordsPerHllpp
+      hllppArray(hllppIndex).update(LongArrayInput(buffer), offset, value, child.dataType)
```

Comment:
you can just pass `InternalRow(buffer)` here, to save a lot of code changes. If performance matters here, you can create a `LongArrayInternalRow` to avoid boxing.

Reply:
InternalRow(buffer) will copy the buffer.
Creating a LongArrayInternalRow is a good idea, thanks!